File: 2024_04/0406/beginner/change.cpp
#include <iostream>

using namespace std;

int main() {
    int defaultMoney = 1000;
    int changesMoney = 0;
    int count = 0;
    int i = 0;
    int coin[6] = {500, 100, 50, 10, 5, 1};
    cin >> changesMoney;
    defaultMoney -= changesMoney;

    while (defaultMoney > 0) {
        if (defaultMoney >= coin[i]) {
            defaultMoney -= coin[i];
            count++;
        } else {
            i++;
        }
    }

    cout << count;
    return 0;
}

File: 2024_04/0406/beginner/change.py
default_money = 1000
changes_money = 0
count = 0
i = 0
coin = [500, 100, 50, 10, 5, 1]

changes_money = int(input())
default_money -= changes_money

while default_money > 0:
    if default_money >= coin[i]:
        default_money -= coin[i]
        count += 1
    else:
        i += 1

print(count)
